File: sorting-algos/insertion_sort/insertion_sort.py
def insertion_sort(arr,size):
    i=1             #array with one element is already sorted so i starts with 1
    while(i<size):
        j=i-1
        key=arr[i]
        while(key<arr[j] and j>=0): #loop inserts key in the already sorted array
            arr[j+1]=arr[j]
            arr[j]=key
            j=j-1
        i=i+1

if(__name__=='__main__'):
    arr=[]
    size=int(input('Enter the number of elements in array\n'))
    print('Enter the elements')
    for i in range(size):
        temp=int(input())
        arr.append(temp)
    insertion_sort(arr,size)
    print('The array after sorting is')
    print(arr)

File: sorting-algos/insertion_sort/insertion_sort.c
// Insertion Sort
#include<stdio.h>
void main()
{
    int n,i, j, key;
    printf("Enter Size of an Array : ");
    scanf("%d",&n);
    int a[n];
    for ( i=0; i< n; i++)                                 // Input in Array
    {
        printf("Enter Element Number %d :",(i+1));
        scanf("%d",&a[i]);
    }
    for ( i=1; i< n; i++ )
    {
        key = a[i];                        //Starts from 2nd element of array a.
        j = i-1;                           // j is the previous element.
        while ( a[j]>=key && j>=0 )          // If previous elements are greater and pointer is not at the end.
        {
            a[j+1] = a[j];                 // Shift all the elements to right side.
            j -= 1;                        // Decrement j
        }
        a[j+1] = key;                        // return key to its new position.
    }
    printf("Sorted Array --> \n");
    for ( i=0; i<n; i++ )
        printf("%d ",a[i]);
}
